PowerEditor/src/MISC/PluginsManager/PluginsManager.cpp

```diff
@@ -76,40 +76,38 @@ bool PluginsManager::unloadPlugin(int index, HWND nppHandle)
     return true;
 }
 
-static WORD GetBinaryArchitectureType(const TCHAR *filePath)
+static WORD getBinaryArchitectureType(const TCHAR *filePath)
 {
-	WORD machine_type = IMAGE_FILE_MACHINE_UNKNOWN;
-	HANDLE hMapping = NULL;
-	LPVOID addrHeader = NULL;
-
 	HANDLE hFile = CreateFile(filePath, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_READONLY, NULL);
 	if (hFile == INVALID_HANDLE_VALUE)
-		goto cleanup;
+	{

```

Comment:
@donho instead of multiple return statement why not single return. e.g. put everything in try and through the result, catch the result and return it. This way multiple `CloseHandle` can be avoided. Just a thought.

Reply:
It's another way to deal with error.
Since it's a small function, I'll keep this way. Thank you for the suggestion.